pkg/infrastructure/clusterapi/ignition.go

```diff
@@ -141,26 +158,52 @@ func addLoadBalancersToInfra(platform string, config *igntypes.Config, publicLBs
 
 func updatePointerIgnition(in IgnitionInput, privateLBs []string, role string) ([]byte, error) {
 	ignData := &igntypes.Config{}
-	err := json.Unmarshal(in.MasterIgnData, ignData)
-	if err != nil {
-		return nil, fmt.Errorf("failed to unmarshal master ignition: %w", err)
-	}
-	if len(privateLBs) > 0 {
-		ignitionHost := net.JoinHostPort(privateLBs[0], "22623")
-		ignData.Ignition.Config.Merge[0].Source = ignutil.StrToPtr(func() *url.URL {
-			return &url.URL{
-				Scheme: "https",
-				Host:   ignitionHost,
-				Path:   fmt.Sprintf("/config/%s", role),
+	switch role {
+	case "master":
+		err := json.Unmarshal(in.MasterIgnData, ignData)
+		if err != nil {
+			return nil, fmt.Errorf("failed to unmarshal master ignition: %w", err)
+		}
+		if len(privateLBs) > 0 {
+			ignitionHost := net.JoinHostPort(privateLBs[0], "22623")
+			ignData.Ignition.Config.Merge[0].Source = ignutil.StrToPtr(func() *url.URL {
+				return &url.URL{
+					Scheme: "https",
+					Host:   ignitionHost,
+					Path:   fmt.Sprintf("/config/%s", role),
+				}
+			}().String())
+			editedIgnBytes, err := json.Marshal(ignData)
+			if err != nil {
+				return nil, fmt.Errorf("failed to convert master pointer ignition data to json: %w", err)
 			}
-		}().String())
-		editedIgnBytes, err := json.Marshal(ignData)
+			return editedIgnBytes, nil
+		}
+		return in.MasterIgnData, nil
+	case "worker":
+		err := json.Unmarshal(in.WorkerIgnData, ignData)
 		if err != nil {
-			return nil, fmt.Errorf("failed to convert pointer ignition data to json: %w", err)
+			return nil, fmt.Errorf("failed to unmarshal worker ignition: %w", err)
+		}
+		if len(privateLBs) > 0 {
```

Comment:
This could be made into a small function right so we don't duplicate the code above.

Reply:
Removed duplication.